packages/knip/src/plugins/nuxt/index.ts

```diff
@@ -24,37 +25,140 @@ const production = [
 
-  const patterns = [
-    'app.{vue,jsx,tsx}',
-    'error.{vue,jsx,tsx}',
-    join(typeof localConfig.dir?.pages === 'string' ? localConfig.dir.pages : 'pages', '**/*.{vue,jsx,tsx}'),
-    join(typeof localConfig.dir?.layouts === 'string' ? localConfig.dir.layouts : 'layouts', '**/*.{vue,jsx,tsx}'),
-    join(typeof localConfig.dir?.middleware === 'string' ? localConfig.dir.middleware : 'middleware', '**/*.ts'),
-    'server/api/**/*.ts',
-    'server/routes/**/*.ts',
-    'server/middleware/**/*.ts',
-    'server/plugins/**/*.ts',
-  ].map(pattern => toProductionEntry(join(srcDir, pattern)));
+  for (const layer of nuxt.options._layers) {
+    const config = layer.cwd === nuxt.options.rootDir ? nuxt.options : (layer.config as NuxtOptions);
+    const srcDir = layer.config.srcDir || layer.cwd;
+    const rootDir = layer.cwd;
+
+    const middlewareDir = resolve(srcDir, resolveAlias(config.dir?.middleware || 'middleware', nuxt.options.alias));
+    const pluginsDir = resolve(srcDir, resolveAlias(config.dir?.plugins || 'plugins', nuxt.options.alias));
+    const serverDir = resolve(srcDir, resolveAlias(config.serverDir || 'server', nuxt.options.alias));
+
+    const entryPatterns: string[] = [
+      // nitro routes
+      resolve(serverDir, config.nitro.apiDir || 'api', `**/*{${extensionGlob}}`),
+      resolve(serverDir, config.nitro.routesDir || 'routes', `**/*{${extensionGlob}}`),
+      resolve(serverDir, `middleware/**/*{${extensionGlob}}`),
+      resolve(serverDir, `plugins/**/*{${extensionGlob}}`),
+      resolve(serverDir, `tasks/**/*{${extensionGlob}}`),
+      // nuxt app
+      resolve(srcDir, resolveAlias(config.dir?.layouts || 'layouts', nuxt.options.alias), `**/*{${extensionGlob}}`),
+      join(middlewareDir, `*{${extensionGlob}}`),
+      join(middlewareDir, `*/index{${extensionGlob}}`),
+      join(pluginsDir, `*{${extensionGlob}}`),
+      join(pluginsDir, `*/index{${extensionGlob}}`),
+      join(srcDir, `app{${extensionGlob}}`),
+      join(srcDir, `App{${extensionGlob}}`),
+    ];
+
+    // file-system routing integration with vue-router
+    if (isPagesEnabled) {
+      for (const pattern of pagesPatterns) {
+        entryPatterns.push(resolve(srcDir, resolveAlias(config.dir?.pages || 'pages', nuxt.options.alias), pattern));
+      }
+    }
+
+    // add nitro auto-imported paths to cover edge case where they are
+    // enabled for nitro but not for nuxt
+    if (isNitroImportsEnabled) {
+      availableSources.add(resolve(rootDir, config.dir.shared ?? 'shared', `utils/*{${extensionGlob}}`));
+      availableSources.add(resolve(rootDir, config.dir.shared ?? 'shared', `types/*{${extensionGlob}}`));
+    }
+
+    inputs.push(...entryPatterns.map(s => toProductionEntry(s)));
+  }
+
+  // TODO: we need to register these as source files which may not be all used
+  for (const path of availableSources) {
+    inputs.push(toProductionEntry(path, { allowIncludeExports: true }));
+  }
```

Comment:
is there a way to register these paths so any imports _into_ them are respected, but unused exports are detected?

Reply:
Yes, if I understand correctly, this is the breakdown:

- default behavior: exports of entry files are not reported
- `skipExportsAnalysis: false` - report unused exports (what you're asking for)
- `allowIncludeExports: true` - report unused exports only with flag `--include-entry-exports`/`includeEntryExports`